test/GPv2Settlement/Swap/Balances.t.sol

```diff
@@ -0,0 +1,116 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+pragma solidity ^0.8;
+
+import {Vm} from "forge-std/Test.sol";
+
+import {GPv2Order, GPv2Signing, IERC20, IVault} from "src/contracts/GPv2Settlement.sol";
+
+import {Helper} from "../Helper.sol";
+import {SwapEncoder} from "test/libraries/encoders/SwapEncoder.sol";
+
+contract Balances is Helper {
+    using SwapEncoder for SwapEncoder.State;
+
+    function test_performs_a_swap_from_erc20_to_erc20_balance_when_specified() public {
+        performs_a_swap_with_the_specified_balances(GPv2Order.BALANCE_ERC20, GPv2Order.BALANCE_ERC20);
+    }
+
+    function test_performs_a_swap_from_internal_to_erc20_balance_when_specified() public {
+        performs_a_swap_with_the_specified_balances(GPv2Order.BALANCE_INTERNAL, GPv2Order.BALANCE_ERC20);
+    }
+
+    function test_performs_a_swap_from_erc20_to_external_balance_when_specified() public {
+        performs_a_swap_with_the_specified_balances(GPv2Order.BALANCE_ERC20, GPv2Order.BALANCE_EXTERNAL);
+    }
+
+    function test_performs_a_swap_from_internal_to_external_balance_when_specified() public {
+        performs_a_swap_with_the_specified_balances(GPv2Order.BALANCE_INTERNAL, GPv2Order.BALANCE_EXTERNAL);
+    }
+
+    function test_performs_a_swap_from_erc20_to_internal_balance_when_specified() public {
+        performs_a_swap_with_the_specified_balances(GPv2Order.BALANCE_ERC20, GPv2Order.BALANCE_INTERNAL);
+    }
+
+    function test_performs_a_swap_from_internal_to_internal_balance_when_specified() public {
+        performs_a_swap_with_the_specified_balances(GPv2Order.BALANCE_INTERNAL, GPv2Order.BALANCE_INTERNAL);
+    }
```

Comment:
nit: I do find it a little odd to have the `buyTokenBalance` defined first, as opposed to `sellTokenBalance` (given the ordering seen in most places for the `Order` struct, the logical path when conisidering token flow, and the definition in the original tests being `sellTokenBalance` first).

Reply:
And on second look, the test labelling is reversed when considering from the perspective of token flow.

Reply:
I inverted the two.
Indeed in hindsight I'd invert the standard ordering to be buy then sell. Still, I've noticed when writing this test that I actually mixed up the two in the function input and only noticed and fixed it when running the test. So being consistent is better here.